app/src/main/java/com/example/weather/network/Dao.kt

```diff
@@ -21,8 +21,8 @@ interface CityOverviewDao {
 @Dao
 interface CityDetailsDao {
 
-    @Query("SELECT * FROM city_details_table where cityId = cityId")
-    fun getCityData(cityId: String): List<CachedCityDetails>
+    @Query("SELECT * FROM city_details_table WHERE id = :cityId")
```

Comment:
this should say `WHERE cityId = :cityId"` it has to match the field in your Cached data class